src/input.cpp

```diff
@@ -0,0 +1,396 @@
+EXTERN int luasteam_getInputTypeForHandle(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    lua_pushstring(L, steam_input_types[SteamInput()->GetInputTypeForHandle(inputHandle)]);
+    return 1;
+}
+
+// InputMotionData_t GetMotionData( InputHandle_t inputHandle );
+EXTERN int luasteam_getMotionData(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    InputMotionData_t mot = SteamInput()->GetMotionData(inputHandle);
+    lua_createtable(L, 0, 10);
+    lua_pushnumber(L, mot.rotQuatX);
+    lua_setfield(L, -2, "rotQuatX");
+    lua_pushnumber(L, mot.rotQuatY);
+    lua_setfield(L, -2, "rotQuatY");
+    lua_pushnumber(L, mot.rotQuatZ);
+    lua_setfield(L, -2, "rotQuatZ");
+    lua_pushnumber(L, mot.rotQuatW);
+    lua_setfield(L, -2, "rotQuatW");
+    lua_pushnumber(L, mot.posAccelX);
+    lua_setfield(L, -2, "posAccelX");
+    lua_pushnumber(L, mot.posAccelY);
+    lua_setfield(L, -2, "posAccelY");
+    lua_pushnumber(L, mot.posAccelZ);
+    lua_setfield(L, -2, "posAccelZ");
+    lua_pushnumber(L, mot.rotVelX);
+    lua_setfield(L, -2, "rotVelX");
+    lua_pushnumber(L, mot.rotVelY);
+    lua_setfield(L, -2, "rotVelY");
+    lua_pushnumber(L, mot.rotVelZ);
+    lua_setfield(L, -2, "rotVelZ");
+    return 1;
+}
+
+// const char * GetStringForActionOrigin( EInputActionOrigin eOrigin );
+EXTERN int luasteam_getStringForActionOrigin(lua_State *L) {
+    EInputActionOrigin eOrigin = static_cast<EInputActionOrigin>(luaL_checkoption(L, 1, nullptr, input_action_origins));
+    lua_pushstring(L, SteamInput()->GetStringForActionOrigin(eOrigin));
+    return 1;
+}
+
+// void RunFrame();
+EXTERN int luasteam_runFrame(lua_State *L) {
+    SteamInput()->RunFrame();
+    return 1;
+}
+
+// void SetLEDColor( InputHandle_t inputHandle, uint8 nColorR, uint8 nColorG, uint8 nColorB, unsigned int nFlags );
+EXTERN int luasteam_setLEDColor(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    uint8 nColorR = luaL_checkinteger(L, 2);
+    uint8 nColorG = luaL_checkinteger(L, 3);
+    uint8 nColorB = luaL_checkinteger(L, 4);
+    bool resetColor = lua_toboolean(L, 5);
+    SteamInput()->SetLEDColor(inputHandle, nColorR, nColorG, nColorB, resetColor ? k_ESteamControllerLEDFlag_RestoreUserDefault : k_ESteamControllerLEDFlag_SetColor);
+    return 1;
+}
+
+// bool ShowBindingPanel( InputHandle_t inputHandle );
+EXTERN int luasteam_showBindingPanel(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    lua_pushboolean(L, SteamInput()->ShowBindingPanel(inputHandle));
+    return 1;
+}
+
+// void StopAnalogActionMomentum( InputHandle_t inputHandle, InputAnalogActionHandle_t eAction );
+EXTERN int luasteam_stopAnalogActionMomentum(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    uint64 eAction = luasteam::checkuint64(L, 2);
+    SteamInput()->StopAnalogActionMomentum(inputHandle, eAction);
+    return 1;
```

Comment:
Should also be 0. I won't comment for all the cases, check the functions and always return the number of return values.